stable_baselines3/common/evaluation.py

```diff
@@ -49,15 +58,32 @@ def evaluate_policy(
         episode_length = 0
         while not done:
             action, state = model.predict(obs, state=state, deterministic=deterministic)
-            obs, reward, done, _info = env.step(action)
+            obs, reward, done, info = env.step(action)
             episode_reward += reward
             if callback is not None:
                 callback(locals(), globals())
             episode_length += 1
             if render:
                 env.render()
-        episode_rewards.append(episode_reward)
-        episode_lengths.append(episode_length)
+
+        # Remove VecEnv stacking (if any)
+        if isinstance(env, VecEnv):
+            info = info[0]
+
+        if "episode" in info.keys():
+            # Monitor wrapper includes "episode" key in info if environment
+            # has been wrapped with it. Use those rewards instead.
+            episode_rewards.append(info["episode"]["r"])
+            episode_lengths.append(info["episode"]["l"])
+        else:
+            episode_rewards.append(episode_reward)
+            episode_lengths.append(episode_length)
+            warnings.warn(
```

Comment:
we should only warn once, no?

Reply:
`warnings`, by default, suppresses further warnings from same code location so only one is shown, but we can add a simple if-check if you feel that would be less stressful for machine.